### PR TITLE
Add contracts

### PR DESCRIPTION
**Pull request by 0xSachinK**

### CODE REVIEW

File: contracts/EuroPool.sol [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    orders[numOrders] = Order({
+      id: numOrders,
+      amount: amount,
+      offChainPaymentAddress: offChainPaymentAddress,
+      claimed: false,
+      completedAt: 0,
+      takerEmailHash: [uint256(0), uint256(0), uint256(0)]
+    });
+
+    ageuro.transferFrom(msg.sender, address(this), amount);
```
Comment (felix2feng) [inline contracts/EuroPool.sol]: Security: Do transfers first, then create order

File: contracts/EuroPool.sol [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+    numOrders += 1;
+  }
+
+  // Called by the taker
+  // Claims an order in the pool
+  function claimOrder(
+    uint256 id,
+    bytes32 emailHash // will be set to 0 for the demo
+  ) external override {
+    require(!orders[id].claimed, "Order has already been claimed.");
```
Comment (felix2feng) [inline contracts/EuroPool.sol]: Optional: Require orderId < numOrders or that the order exists

File: contracts/EuroPool.sol [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+    uint256[3] memory senderEmailHash = [inputs[5], inputs[6], inputs[7]];
+
+    require(
+      orders[orderId].claimed,
+      "Order must be claimed before it can be completed."
+    );
+    require(orders[orderId].completedAt == 0, "Order must not be completed.");
+
+    // verify proof
+    verifier.verify(proof.a, proof.b, proof.c, inputs);
```
Comment (felix2feng) [inline contracts/EuroPool.sol]: I assume this reverts if it fails vs. return true/false?

File: contracts/EuroPool.sol [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    bytes32 emailHash // will be set to 0 for the demo
+  ) external override {
+    require(!orders[id].claimed, "Order has already been claimed.");
+    claimedOrders[id] = emailHash;
+    orders[id].claimed = true;
+  }
+
+  // Called by the taker
+  // Completes an order in the pool
+  function completeOrder(
```
Comment (felix2feng) [inline contracts/EuroPool.sol]: For our demo, do we even need the completeOrder method? We just need the off-ramp method.

**Comment by 0xSachinK** [comment]
@felix2feng Lets do a high-level review of tests and then we can merge.

**Pull request merged**